### ISSUE TITLE
Test Open/System One as cheap prospective pattern completion

### ISSUE THREAD

**Issue by rinsakamo** [rinsakamo/relay-self#268]
## Summary

Test the first concrete **OPEN / SYSTEM_ONE** candidate as cheap prospective pattern completion:

~~~text
current Present
+ selected prior episode Memory
        ↓
OPEN / S1
one short non-enumerated expected-evidence hypothesis
        ↓
later grounded World observation
        ↓
MATCH | MISMATCH | UNKNOWN
~~~

The goal is to test whether a cheap open proposal can contribute useful candidate information before deliberate open generation, without converting the task into a finite-choice classification problem.

This is a research / falsification ledger. It is not semantic authority and does not authorize a new cognition owner.

Parent: #266.
Related: #265 #267.

## Fresh repository state at creation

~~~text
main = 6e9a302f6796479e319a35a5d1c6837cceb95edd
tree = c11550360eb5cb65c6fff05096ec12e2482e69ac
open PRs targeting main = 0
ruleset 23442682 = active
~~~

Current authority always supersedes these recorded hashes.

## Why this candidate

The first tempting candidate was to generate current experiment-local ExpectedEvidence such as:

~~~text
arrived:cave = true
route_open:cave = true
movement_progress = true
~~~

That failed Grand Null.

Those expectations are currently either direct consequences of the selected path or explicit assumptions already present in Present/Action semantics.

Therefore:

~~~text
regenerating an already-known Action/Present expectation
!= independent Open/S1 cognition
~~~

The first useful surface must require a candidate that is not already encoded as an Action postcondition or finite answer set.

## Target property

The proposal request must not enumerate admissible outcomes.

Candidate output may have a small structural envelope such as:

~~~text
expected evidence:
  key: <free candidate key>
  value: <free JSON value>
~~~

but the set of possible key/value content must not be supplied as finite choices.

Thus:

~~~text
typed envelope != closed answer space
~~~

The cognition step is OPEN because it must supply candidate semantic content.

## First apparatus

Prefer an experiment-local symbolic / Minecraft-like trajectory fixture with:

- Present prefix;
- selected prior episode Memory;
- later observed consequence held out from proposal generation;
- opaque/remapped event identifiers where practical;
- no human labels such as danger, fear, safety, death, success, failure in the proposal input unless they are part of an explicitly separate semantic-control condition.

Example conceptual shape:

~~~text
Memory episode:
  p0 -> p1 -> event:kappa

Current episode:
  q0 -> q1 -> ?

OPEN/S1 proposal:
  expected = event:lambda

Later grounded observation:
  event:lambda
~~~

The exact fixture should not make the answer a trivial lexical copy.

If a simple deterministic pattern-completion rule solves the fixture completely, use that rule and reject the need for model-based Open/S1.

## Grounded evaluation

Reuse the existing consequence distinction where practical:

~~~text
ExpectedEvidence
ObservedEvidence
compare_consequence(...)
  -> MATCH
  -> MISMATCH
  -> UNKNOWN
~~~

Important:

- generated expectation is not World truth;
- later observed evidence remains authoritative for the comparison;
- mismatch must not rewrite the observation;
- unknown consequence must not be relabeled into success/failure;
- no automatic Memory/training label follows merely from a generated proposal.

This provides a deterministic downstream judge rather than an LLM-as-judge for the primary outcome.

## Candidate matched conditions

After deterministic apparatus qualification, compare at least:

### A. Grand-Null baseline

Cheapest deterministic mechanism that can use the same allowed information.

This may be:

- no proposal;
- exact repeat detection;
- deterministic trajectory rule;
- existing Retrieval if it fully reduces the task.

### B. OPEN / S1 candidate

One short proposal with:

- same Present + selected Memory;
- no finite options;
- no explanation requirement;
- bounded small output budget;
- no hidden retry/fallback;
- no semantic authority.

The exact physical mechanism is intentionally not fixed before #267 clarifies the shared generation substrate.

### C. OPEN / S2 baseline

A deeper generated open-cognition path over the same semantic input, allowed a larger cognition/output budget.

The comparison asks whether the cheap candidate preserves enough useful prospective signal at materially lower cost.

## Optional CLOSED/S1 follow-up

Do not force this into the first gate.

If the Open/S1 proposal is useful, a later composition may test:

~~~text
OPEN/S1
  -> candidate hypothesis
  -> CLOSED/S1
       supported | rejected | unresolved
~~~

before that hypothesis affects Attention, Skill pressure, or planning.

That verification step must remain separate from the generation of the candidate.

## Anti-leakage controls

The first useful experiment should prevent easy semantic cheating.

Prefer, where practical:

- opaque event ids;
- mapping permutation across cases;
- blinded request/provenance identifiers;
- no expected outcome in request id;
- no future frame in proposal input;
- no finite hidden list serialized into the prompt;
- held-out trajectory suffix;
- same prompt/state surface across matched execution conditions.

Record any remaining pretrained semantic prior explicitly.

## Metrics

Keep cognition utility and cost separate.

Record:

- proposal validity;
- exact grounded MATCH / MISMATCH / UNKNOWN when applicable;
- abstention/unresolved behavior if supported;
- hallucinated / unparsable proposal rate;
- prompt tokens;
- completion tokens;
- wall latency;
- provider-call count;
- retry/replay count;
- whether deterministic baseline already solves the task;
- whether Open/S1 materially changes downstream usable information;
- whether Open/S2 gains quality at additional cost.

Do not collapse these into one universal score.

### F1 — deterministic reduction

The task is fully served by existing Retrieval, exact pattern matching, an Action postcondition, or another deterministic mechanism.

Result:

~~~text
Open/S1 not justified on this surface
~~~

### F2 — short open generation is merely truncated bad S2

The cheap path loses essential correctness or produces unstable/unusable candidates without a distinct useful operating region.

Do not call shorter output System One merely because it is faster.

### F3 — semantic leakage dominates

Opaque/remapped controls fail while semantic labels succeed.

Record pretrained semantic dependence, not general prospective cognition.

### F4 — candidate adds no downstream information

The proposal can be generated but does not improve later Attention/verification/action-relevant cognition beyond available state.

Keep it as narration or reject it.

### F5 — useful cheap open proposal

A non-enumerated candidate is grounded by later evidence with useful reliability and materially lower cost than the deeper open path, while deterministic baselines do not already solve the task.

This would justify a stronger Open/S1 implementation transaction.

## Relation to #265

If this mechanism survives independently, #265 may later supply a more natural input surface:

~~~text
unlabeled recurrent value-gradient pattern
  -> cheap resemblance / prospective proposal
~~~

Do not use #265 emotion labels to make this first experiment easier.

## Relation to #267

#267 owns the shared System-One/System-Two execution substrate.

This Issue defines an evaluation surface, not a new provider.

Preserve:

~~~text
one RelayEngine
one cognition state source
different execution cost/regime only where evidence justifies it
~~~

## Grand Null

Do not add:

- OpenThinking runtime owner;
- intuition owner;
- imagination owner;
- prospective-cognition owner;
- universal prediction service;
- per-tick model call;
- new Memory owner;
- hidden finite answer list;
- LLM judge as primary ground truth;
- automatic training from generated hypotheses;
- cache reuse claim.

## Completion

Complete when:

1. a deterministic apparatus demonstrates a genuinely non-enumerated prospective target without future leakage;
2. deterministic reduction is tested first;
3. if an independent model question survives, one matched physical comparison tests cheap open proposal vs deeper open generation;
4. the result is classified under the falsifiers above;
5. any runtime promotion occurs only in a separate bounded transaction.

Core question:

> Can RelaySelf cheaply produce a useful "これ次に○○じゃない？" candidate before deliberate open cognition, and have the World later prove it right, wrong, or unknown?

**Comment by rinsakamo** [comment]
## Deterministic apparatus merged; model/system comparison remains

PR #269 merged as:

~~~text
main = caacb0ca82a10316ad9eeb8d6bdfa8d440a82dd0
tree = 01f346334b9632c2ca41a6ea0fcb6dc8ef7630a8
~~~

Exact PR head `ce49f9669eb1a4b3bbb9034a1405a08bbf72aa9d` passed:

- repository-contracts
- pytest
- lint
- mineflayer-adapter

The deterministic apparatus now proves only the evaluation shape:

~~~text
current prefix + selected Memory
  -> OpenCognitionRequest with no finite choices
  -> held-out future excluded from request
  -> experiment-local {key,value} proposal
  -> later grounded evidence
  -> MATCH | MISMATCH | UNKNOWN
~~~

Additional guards establish:

- changing only held-out future evidence does not change the proposal request;
- visible future leakage fails closed;
- Present-like and Memory-like provenance remain distinct;
- the trivial exact-replay baseline does not solve the reference surface;
- no src/relay_self or semantic authority changed.

This does **not** prove useful Open/S1 model behavior.

Remaining work is now narrower:

1. select the cheapest plausible Open/S1 physical mechanism after #267 clarifies the shared execution substrate;
2. define a matched deeper Open/S2 baseline over the same semantic input;
3. run the model/system comparison only after deterministic request identity / anti-leakage preflight;
4. classify against F1-F5 without promoting runtime semantics from a preferred result.

Keep #268 OPEN.
